### ISSUE TITLE
Can we get a more stripped down npm package?

### ISSUE THREAD

**Issue by TooTallNate** [justjake/quickjs-emscripten#117]
The [published package on npm is a whopping 8.62mb](https://packagephobia.com/result?p=quickjs-emscripten@0.23.0). That's a heavy ask for packages that are trying to be size-conscious, like CLI programs. Seems like a lot of the package is variations of the WASM binary (debug vs. release mode, sync vs. async mode, etc.).

Not sure what the best solution here is, but personally I am only interested in the release/sync variant.

**Comment by justjake** [comment]
I think we could put each build in its own NPM package and have each runtime import the support library. I figure in the mean time a Webpack or similar build will eliminate unused code from your distributable.

**Comment by fvictorio** [comment]
@justjake are you familiar with the approach that [`napi-rs`](https://github.com/napi-rs/napi-rs) takes?

Basically, you publish a single slim package with optional dependencies that look like this:

```
  "optionalDependencies": {
    "@my-org/my-package-darwin-arm64": "0.1.1",
    "@my-org/my-package-win32-arm64-msvc": "0.1.1",
    "@my-org/my-package-linux-arm64-gnu": "0.1.1",
    "@my-org/my-package-linux-arm64-musl": "0.1.1",
    "@my-org/my-package-win32-ia32-msvc": "0.1.1",
    "@my-org/my-package-darwin-x64": "0.1.1",
    "@my-org/my-package-win32-x64-msvc": "0.1.1",
    "@my-org/my-package-linux-x64-gnu": "0.1.1",
    "@my-org/my-package-linux-x64-musl": "0.1.1",
    "@my-org/my-package-freebsd-x64": "0.1.1"
  }
```

and then the `index.js` file does something like this:

```js
switch (platform) {
  case "android":
    switch (arch) {
      case "arm64":
        localFileExisted = existsSync(
          join(__dirname, "my-package.android-arm64.node")
        );
        try {
          if (localFileExisted) {
            nativeBinding = require("./my-package.android-arm64.node");
          } else {
            nativeBinding = require("@my-org/my-package-android-arm64");
          }
        } catch (e) {
          loadError = e;
        }
        break;
      case "arm":
        localFileExisted = existsSync(
          join(__dirname, "my-package.android-arm-eabi.node")
        );
        try {
          if (localFileExisted) {
            nativeBinding = require("./my-package.android-arm-eabi.node");
          } else {
            nativeBinding = require("@my-org/my-package-android-arm-eabi");
          }
        } catch (e) {
          loadError = e;
        }
        break;
      default:
        throw new Error(`Unsupported architecture on Android ${arch}`);
    }
    break;
  case "win32":
    // ...
```

**Comment by justjake** [comment]
That doesn’t sound like it would work well for browsers though :-/

**Comment by justjake** [comment]
I Guess you have another branch for “browser”?

**Comment by fvictorio** [comment]
Ah, N-API only works for node.js, you are right. But I'm pretty sure there's something that can be done to add browser builds too (perhaps using the `browser` field in the `package.json`)

**Comment by justjake** [comment]
The issue here is that we have 4 web compatible variants, and there’s no way to automatically understand what the user wants. We’d need to do sub-packages and optional peer deps and ask the user to choose their preferred variant. 

I’m happy to accept a PR to split things up, but don’t have time to work on it myself.

**Comment by justjake** [comment]
@TooTallNate this is fixed as of v0.25.0. I added a new package quickjs-emscripten-core for your use-case. See this section of the README: https://github.com/justjake/quickjs-emscripten/tree/main#reducing-package-size

**Comment by justjake** [comment]
Let me know if it works for you :)
